app/src/main/res/layout/meme_card_detail_layout.xml

```diff
@@ -0,0 +1,105 @@
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent" />
+
+    <TextView
+        android:id="@+id/meme_top_text_title"
+        style="@style/Letters"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/title"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/meme_image" />
+
+    <TextView
+        android:id="@+id/meme_top_text_description"
+        style="@style/Body"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/meme_top_text_title" />
+
+    <TextView
+        android:id="@+id/meme_button_text_title"
+        style="@style/Letters"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/other_names"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/meme_top_text_description" />
+
+    <TextView
+        android:id="@+id/meme_button_text_description"
+        style="@style/Body"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/meme_button_text_title" />
+
+    <TextView
+        android:id="@+id/meme_rank_title"
+        style="@style/Letters"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/ranking"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/meme_button_text_description" />
+
+    <TextView
+        android:id="@+id/meme_rank_description"
+        style="@style/Body"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintLeft_toLeftOf="@id/meme_button_text_description"
+        app:layout_constraintTop_toBottomOf="@id/meme_rank_title" />
+
+    <TextView
+        android:id="@+id/meme_tags_title"
+        style="@style/Letters"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/tags"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/meme_rank_description" />
+
+    <TextView
+        android:id="@+id/meme_tags_description"
+        style="@style/Body"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/meme_tags_title" />
+
+    <ProgressBar
+        android:id="@+id/progress_bar"
+        android:layout_width="70dp"
+        android:layout_height="70dp"
```

Comment:
move this hardcoded values into the dimens file

Reply:
Done thx